.github/workflows/main.yml

```diff
@@ -36,6 +36,10 @@ jobs:
       - name: Build
         run: make -C $GITHUB_WORKSPACE all
       - name: Test
-        run: go test -v ./...
+        run: go test -v -covermode=atomic -coverprofile=coverage.out ./...
+      - name: Upload Coverage Report
+        uses: codecov/codecov-action@f32b3a3741e1053eb607407145bc9619351dc93b # v2.1.0
```

Comment:
Is this isolated such that if codecov was compromised in the same way that it was previously compromised, the action couldn't exfiltrate secrets from our workflow?

Reply:
Someone would need to add a secret to this workflow (there are currently none) or add additional permissions to the github token to do anything as I understand it

Reply:
FWIW codecov completely rewrote the uploader so the attack vector would also have to be different this time as well

Reply:
For some background, CodeCov also changed their uploader significantly after the incident. Its no longer a bash script that you download via curl. Its now a static binary. This action downloads the binary and checks the signature using their signing key.

The manual process to do this is documented here https://docs.codecov.com/docs/codecov-uploader